.eslintrc.cjs

```diff
@@ -172,6 +175,17 @@ module.exports = {
 			},
 		},
 
+		{
+			files: [
+				'**/*.config.{js,ts,cjs,mjs}',
+				'**/config/**/*.{js,ts,cjs,mjs}',
+				'**/.eslintrc.{js,ts,cjs,mjs}',
```

Comment:
1. First rule is too loose. _Any_ file that has the word `.config.` in it? 
2. Anything with `.mjs` or `.ts` supports `import` and should be enforced.
3. If you're matching specific files (like a local eslint file) there's no need to regex a bunch of different file etcs.
4. You anyway might consider the `allow` before using the `overrides` https://typescript-eslint.io/rules/no-require-imports/#allow